awswrangler/dynamodb/_utils.py

```diff
@@ -34,6 +38,103 @@ def get_table(
+    except ClientError as err:
+        if err.response["Error"]["Code"] == "ResourceNotFoundException":
+            _logger.error("Couldn't execute PartiQL: '%s' because the table does not exist.", kwargs["Statement"])
+        else:
+            _logger.error(
+                "Couldn't execute PartiQL: '%s'. %s: %s",
+                kwargs["Statement"],
+                err.response["Error"]["Code"],
+                err.response["Error"]["Message"],
+            )
+        raise
+    return response
+
+
+def _read_execute_statement(
+    kwargs: Dict[str, Union[str, bool, List[Any]]], boto3_session: Optional[boto3.Session]
+) -> Iterator[Dict[str, Any]]:
+    next_token: str = "init_token"  # Dummy token
+    while next_token:
+        response = _execute_statement(kwargs=kwargs, boto3_session=boto3_session)
+        next_token = response.get("NextToken", None)
+        kwargs["NextToken"] = next_token
+        yield response["Items"]
+
+
+def execute_statement(
+    statement: str,
+    parameters: Optional[List[Any]] = None,
+    consistent_read: bool = True,
+    boto3_session: Optional[boto3.Session] = None,
+) -> Optional[Iterator[Dict[str, Any]]]:
+    """Run a PartiQL statement against a DynamoDB table.
+
+    Parameters
+    ----------
+    statement : str
+        The PartiQL statement.
+    parameters : Optional[List[Any]]
+        The list of PartiQL parameters. These are applied to the statement in the order they are listed.
+    consistent_read: bool
+        The consistency of a read operation. If `True` (by default), then a strongly consistent read is used.
+    boto3_session : Optional[boto3.Session]
+        Boto3 Session. If None, the default boto3 Session is used.
+
+    Returns
+    -------
+    Optional[Iterator[Dict[str, Any]]]
+        An iterator of the items from the statement response, if any.
+
+    Examples
+    --------
+    Insert an item
+
+    >>> import awswrangler as wr
+    >>> wr.dynamodb.execute_statement(
+    ...     statement="INSERT INTO movies VALUE {'title': ?, 'year': ?, 'info': ?}",
+    ...     parameters=[title, year, {"plot": plot, "rating": rating}],
+    ... )
+
+    Select items
+
+    >>> wr.dynamodb.execute_statement(
+    ...     statement="SELECT * FROM movies WHERE title=? AND year=?",
+    ...     parameters=[title, year],
+    ... )
+
+    Update items
+
+    >>> wr.dynamodb.execute_statement(
+    ...     statement="UPDATE movies SET info.rating=? WHERE title=? AND year=?",
+    ...     parameters=[rating, title, year],
+    ... )
+
+    Delete items
+
+    >>> wr.dynamodb.execute_statement(
+    ...     statement="DELETE FROM movies WHERE title=? AND year=?",
+    ...     parameters=[title, year],
+    ... )
+    """
+    kwargs: Dict[str, Union[str, bool, List[Any]]] = {"Statement": statement, "ConsistentRead": consistent_read}
+    if parameters:
+        kwargs["Parameters"] = parameters
+
+    if not statement.startswith("SELECT"):
```

Comment:
Do keywords in PartiQL queries have to be uppercase? Most flavors of SQL don't differentiate between `SELECT`, `select` or `seLeCt`. If that's the same case with PartiQL, we should make sure we cover those cases as well.

We could easily solve this by just doing:
```python
if not statement.upper().startswith("SELECT"):
```

Also, we could add `trim()`, so that we cover the case where the query has leading whitespaces for some reason.